stdlib/src/collections/list.mojo

```diff
@@ -788,7 +788,43 @@ struct List[T: CollectionElement](CollectionElement, Sized, Boolable):
         if i < 0:
             normalized_idx += self[].size
 
-        return (self[].data + normalized_idx)[]
+        return self[].unsafe_get(normalized_idx)
+
+    @always_inline
+    fn unsafe_get[
+        IndexerType: Indexer,
```

Comment:
Note that `Indexer` types can now be implicitly converted to `Int` so such methods were recently reverted to simply accepting `Int` args again https://github.com/modularml/mojo/commit/72c5a47872ffd8ca8e58a827be20da21c5c3d90f

Reply:
Here we have a choice between implicit conversion and explicit conversion. I believe we'll choose explicit to clearly indicate the intent. That should avoid bugs in the future

Reply:
Frankly I agree with you 😄, I just wanted to bring it to your attention in case you hadn't seen it since you'll likely get some pushback from the team.